### ISSUE TITLE
Fixing HtCondor & Pulsar

### ISSUE THREAD

**Issue by Dirowa** [Avans-ATGM/infrastructure#14]

**Comment by Dirowa** [comment]
Some of the new Galaxy server jobs results into a ```condor_submit failed```` error

Causation: new galaxy server refuses to communicate with the old servers

To do:

- [ ] Install Pulsar on the old server
- [ ] Remove Condor entries (*except new server names)
- [ ] Test both servers

**Comment by Dirowa** [comment]
hi @hexylena 

I tried debugging the Pulsar server, which doesn't produce any errors anymore. but how long does it usually takes the first time to finish pulsar job?

**Comment by Dirowa** [comment]
hi @hexylena I,m pretty sure all files are correctly yet i,m still unable to get it working :(

TLDR:
i cant get pulsar working, and it seems like it is stuck but i don't get enough information from the log file. probably same error the galaxy team had last January., Any advice?


I suspect that it lies on the pulsar side on the extra hosts.

```
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283594]: Activating virtual environment /mount/sdb/pulsar/venv/bin/activate
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283594]: Starting pulsar with command [pulsar-main]
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283598]: /mount/sdb/pulsar/venv/lib/python3.8/site-packages/pulsar/util/pastescript/loadwsgi.py:113: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283598]:   return pkg_resources.EntryPoint.parse("x=" + s).load(False)
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283598]: 2021-12-21 14:20:36,895 INFO  [pulsar.core][MainThread] Starting the Pulsar without a toolbox to white-list.Ensure this application is protected by firewall or a configured private token.
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283598]: 2021-12-21 14:20:36,990 INFO  [pulsar.locks][MainThread] pylockfile module not found, skipping experimental lockfile handling.
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283598]: 2021-12-21 14:20:36,991 INFO  [pulsar.messaging.bind_amqp][MainThread] bind_manager_to_queue called for [pyamqp://Isengard:********@galaxy.atgm.avans.nl:5671//pulsar/Isengard?ssl=1] and manager [_default_]
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283598]: 2021-12-21 14:20:36,992 DEBUG [pulsar.client.amqp_exchange][consume-setup-pyamqp://Isengard:********@galaxy.atgm.avans.nl:5671//pulsar/Isengard?ssl=1] Consuming queue '<unbound Queue pulsar__setup -> <unbound Exchange pulsar(direct)> -> pulsar__setup>'
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283598]: 2021-12-21 14:20:36,993 DEBUG [pulsar.client.amqp_exchange][consume-kill-pyamqp://Isengard:********@galaxy.atgm.avans.nl:5671//pulsar/Isengard?ssl=1] Consuming queue '<unbound Queue pulsar__kill -> <unbound Exchange pulsar(direct)> -> pulsar__kill>'
dec 21 14:20:36 galaxy.bioinformatics-atgm.nl pulsar[1283598]: 2021-12-21 14:20:36,994 DEBUG [pulsar.client.amqp_exchange][consume-status-pyamqp://Isengard:********@galaxy.atgm.avans.nl:5671//pulsar/Isengard?ssl=1] Consuming queue '<unbound Queue pulsar__status -> <unbound Exchange pulsar(direct)> -> pulsar__status>'
```

it should also produce the following logs: 
[standard messages as time, host etc. ] AMQP heartbeat thread alive

could it be because of the ``` packages/pulsar/util/pastescript/loadwsgi.py:113: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately. 
return pkg_resources.EntryPoint.parse("x=" + s).load(False)
``` message.

I noticed the great galaxy team already addressed this issue in the pulsar tutorial and roles, but It is there still :(.

what have i tried else:

- installing pulsar manual without success, ``` pulsar.client.exceptions.PulsarClientTransportError: Unknown transport error (transport message: [Errno 111] Connection refused) ```
 - enabling all ports

what do i know for sure:
- jobs are send, in the testing ty go to asgard (according galaxy admin panel and log files of galaxy)
- queuess are empty
- staging directory is empty

so my wild quess currently is that the jobs are send and removed from the queue but that pulsar is not looking for them.
What do you think, or in what direction should i read?

**Comment by Dirowa** [comment]
hi @hexylena,
I have been fiddiling with the Pulsar installation on the galaxy server and tested a few things.
As documented should you use curl to test RabbitMQ. 

but i get these outputs, the one of 5672 is correct as galaxy docs state but does produce a error in the RabbitMQ logs.
```
(base) dirowa@galaxy:~$ curl http://localhost:5672
curl: (1) Received HTTP/0.9 when not allowed
```
Which produces the following in the rabbit MQ log.

```
2022-02-10 10:31:18.861 [info] <0.9400.0> accepting AMQP connection <0.9400.0> (127.0.0.1:2890 -> 127.0.0.1:5672)
2022-02-10 10:31:18.861 [error] <0.9400.0> closing AMQP connection <0.9400.0> (127.0.0.1:2890 -> 127.0.0.1:5672):
{bad_header,<<"GET / HT">>}
2022-02-10 10:31:18.861 [info] <0.9402.0> Closing all channels from connection '127.0.0.1:2890 -> 127.0.0.1:5672' because it has been closed
```

with the  other command produces an different error, but not one in the RabbitMQ logs.
```
(base) dirowa@galaxy:~$ curl -k https://localhost:5671
curl: (35) OpenSSL SSL_connect: Connection reset by peer in connection to localhost:5671
````

I tried solving this online but i wasn't able to find help full information with our settings.
Can i ignore this part or do you have ideas in what part i could look into better?

**Comment by hexylena** [comment]
ok, not surprising the ssl doesn't work, the ssl probabl isn't configured right because of the behaviour of the new role. No, I doubt you'd be able to find information online :( welcome to the final frontier I suppose

**Comment by Dirowa** [comment]
maybe an odd idea @hexylena, but could we not ask the Avans ICT guys to help us with the problem the new servers doesnt want to connect with out other machines bij ssh etc. then we can just use htcondor

**Comment by Dirowa** [comment]
Stale subject due the ports required to be open by DIF team.
needs to be acknoledge then further research needed
